### PR TITLE
NIRES frame typing

### PR DESCRIPTION
**Pull request by kbwestfall**
This PR addresses PN-14 in the Keck/NIRES Jira development path, "As a user, I expect the pipeline to automatically recognize calibration files."  This required no changes to the code base.  This PR only adds the development doc.

### CODE REVIEW

File: doc/dev/niresframes.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      frame types; however, this can be changed using the ``exprng`` parameter
+      in the :ref:`pypeit_file`; see also :ref:`frame_types`.
+
+Testing
+-------
+
+Requirement PN-14 states: "As a user, I expect the pipeline to automatically
+recognize calibration files."
+
+``PypeIt`` meets this requirement as demonstrated by the test at
```
Comment (profxj) [inline doc/dev/niresframes.rst:68]: Nice!

File: doc/dev/niresframes.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The test requires that you have downloaded the ``PypeIt``
+:ref:`dev-suite` and defined the ``PYPEIT_DEV`` environmental
+variable that points to the relevant directory. The algorithm of the
+test is as follows:
+
+    #. Find all the directories in the :ref:`dev-suite` with Keck
+       NIRES data.
+
+    #. For each directory (i.e., instrument setup; current there is only one):
```
Comment (profxj) [inline doc/dev/niresframes.rst]: currently

**Review by profxj — APPROVED** [review]
Great stuff. And I have approved the corresponding
 PR in the DevSuite Repo.

**Review by debora-pe — APPROVED** [review]
Nice docs!!

**Pull request merged**